src/runtime/parcelset/detail/parcel_await.cpp

```diff
@@ -15,31 +15,39 @@
 namespace hpx { namespace parcelset { namespace detail {
 
     parcel_await::parcel_await(parcel&& p, int archive_flags,
-        parcel_await::put_parcel_type pp)
+            parcel_await::put_parcel_type pp)
       : put_parcel_(std::move(pp)),
-        archive_(preprocess_, archive_flags),
-        overhead_(archive_.bytes_written()),
+        archive_flags_(archive_flags),
         idx_(0)
     {
         parcels_.push_back(std::move(p));
+        handled_.push_back(0);
     }
 
     parcel_await::parcel_await(std::vector<parcel>&& parcels, int archive_flags,
-        parcel_await::put_parcel_type pp)
+            parcel_await::put_parcel_type pp)
       : put_parcel_(std::move(pp)),
         parcels_(std::move(parcels)),
-        archive_(preprocess_, archive_flags),
-        overhead_(archive_.bytes_written()),
+        handled_(parcels_.size(), 0),
+        archive_flags_(archive_flags),
         idx_(0)
-    {
-    }
+    {}
 
     void parcel_await::apply()
     {
         for (/*idx_*/; idx_ != parcels_.size(); ++idx_)
         {
-            archive_.reset();
-            archive_ << parcels_[idx_];
+            // send every parcel only once
+            if (handled_[idx_])
+                continue;
+
+            // collect the necessary size the serialization operation
+            hpx::serialization::detail::preprocess preprocess;
```

Comment:
This will not work. preprocess needs to be kept alive as part of parcel_await. Once we have a future we need to wait on, the completion handlers are registered to exactly this object, once we leave the scope in line 63, we will get segfaults due to dangling pointers, for example here:
https://github.com/STEllAR-GROUP/hpx/blob/master/hpx/runtime/serialization/detail/preprocess.hpp#L71
That's also the reason, why we need to keep this alive.